spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/servlet/ServletContextInitializerBeans.java

```diff
@@ -369,6 +370,21 @@ private void configureFromAnnotation(FilterRegistrationBean<Filter> bean, Filter
 			if (registration.urlPatterns().length > 0) {
 				bean.setUrlPatterns(Arrays.asList(registration.urlPatterns()));
 			}
+			for (WebInitParam param : registration.initParameters()) {
+				bean.addInitParameter(param.name(), param.value());
+			}
+
+			@SuppressWarnings("unchecked")
+			Map<String, ServletRegistrationBean<?>> servletRegistrationBeans = (Map<String, ServletRegistrationBean<?>>) (Map<?, ?>) this.beanFactory
```

Comment:
You could consider using the ObjectProvider API to locate all instances of `ServletRegistrationBean`:

```java
this.beanFactory.getBeanProvider(ServletRegistrationBean.class).forEach((servletRegistrationBean) -> {
		for (Class<?> servletClass : registration.servletRegistrationBeans()) {
			if (servletClass.isInstance(servletRegistrationBean.getServlet())) {
				bean.addServletRegistrationBeans(servletRegistrationBean);
			}
		}
});
```